### PR TITLE
Use the real page size instead of hardcoded 4096 bytes that can be false

### PR DESCRIPTION
**Pull request by hmt23**
This PR includes the POSIX way to get the page size of the system.
This PR also includes little fixes like std::size_t instead of size_t and missing override keyword.

### CODE REVIEW

**Comment by hmt23** [comment]
Good catch, Windows is the reason why I only talk about Linux system on my commit message. I was secretly hoping that Gilles would do the job for Windows system :p

**Review by grospelliergilles — CHANGES_REQUESTED** [review]
Thanks for the PR.
Please fix the following points:
- do not mix in the same PR modifications in Arccore and Arcane. The modification in `IMemoryAllocator` is not needed so you can revert it.
- do not use a platform specific call in user code. Add a method to get the page size in `arcane/utils/PlatformUtils.h` (for example `platform::getPageSize()`) and implement it for Linux and Windows with a fallback for other platform (MacOS for example).
- add a test for this call.

**Review by grospelliergilles — COMMENTED** [review]
Thanks for the fix. I will add tests and merge

**Review by grospelliergilles — APPROVED** [review]

File: arcane/src/arcane/utils/PlatformUtils.cc [hunk 1]
```diff
@@ -525,6 +525,25 @@ getRealTimeNS()
...
+extern "C++" ARCANE_UTILS_EXPORT Int64 platform::
+getPageSize()
+{
+#if defined(ARCCORE_OS_WIN32)
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+#elif defined(ARCANE_OS_LINUX)
+  return ::sysconf(_SC_PAGESIZE);
+#else
```
Comment (cedricchevalier19) [inline arcane/src/arcane/utils/PlatformUtils.cc:537]: Not sure why the define names are not very consistent!

Comment (grospelliergilles) [reply arcane/src/arcane/utils/PlatformUtils.cc:537]: Yes you are right. `ARCANE_OS_LINUX` is the same as `ARCCORE_OS_LINUX` but it is not consistent. I will fix it. Thanks

**Pull request merged**